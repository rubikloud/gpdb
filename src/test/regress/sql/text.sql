--
-- TEXT
--

SELECT text 'this is a text string' = text 'this is a text string' AS true;

SELECT text 'this is a text string' = text 'this is a text strin' AS false;

CREATE TABLE TEXT_TBL (f1 text);

INSERT INTO TEXT_TBL VALUES ('doh!');
INSERT INTO TEXT_TBL VALUES ('hi de ho neighbor');

SELECT '' AS two, * FROM TEXT_TBL order by f1;

--
-- TEXT CASTING TO/FROM ANY TYPE
--
SELECT '1'::bool::text;
SELECT array[1,2]::text;
SELECT '{1,2}'::text::integer[];

CREATE TYPE usr_define_type as (id int, name text);
SELECT '(1,abc)'::text::usr_define_type;
