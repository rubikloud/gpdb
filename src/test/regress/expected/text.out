--
-- TEXT
--
SELECT text 'this is a text string' = text 'this is a text string' AS true;
 true 
------
 t
(1 row)

SELECT text 'this is a text string' = text 'this is a text strin' AS false;
 false 
-------
 f
(1 row)

CREATE TABLE TEXT_TBL (f1 text);
INSERT INTO TEXT_TBL VALUES ('doh!');
INSERT INTO TEXT_TBL VALUES ('hi de ho neighbor');
SELECT '' AS two, * FROM TEXT_TBL order by f1;
 two |        f1         
-----+-------------------
     | doh!
     | hi de ho neighbor
(2 rows)

--
-- TEXT CASTING TO/FROM ANY TYPE
--
SELECT '1'::bool::text;
 text
------
 t
(1 row)

SELECT array[1,2]::text;
 array
-------
 {1,2}
(1 row)

SELECT '{1,2}'::text::integer[];
 int4
-------
 {1,2}
(1 row)

CREATE TYPE usr_define_type as (id int, name text);
SELECT '(1,abc)'::text::usr_define_type;
 usr_define_type
-----------------
 (1,abc)
(1 row)

